src/tools/illink/src/ILLink.RoslynAnalyzer/TrimAnalysis/ReflectionAccessAnalyzer.cs

```diff
@@ -191,7 +191,7 @@ void GetDiagnosticsForField (Location location, IFieldSymbol fieldSymbol)
 			if (fieldSymbol.TryGetRequiresUnreferencedCodeAttribute (out var requiresUnreferencedCodeAttributeData))
 				ReportRequiresUnreferencedCodeDiagnostic (location, requiresUnreferencedCodeAttributeData, fieldSymbol);
 
-			if (fieldSymbol.GetDynamicallyAccessedMemberTypes () != DynamicallyAccessedMemberTypes.None) {
+			if (FlowAnnotations.GetFieldAnnotation (fieldSymbol) != DynamicallyAccessedMemberTypes.None) {
```

Comment:
Should we change most of the uses of `ISymbol.GetDynamicallyAccessedMemberTypes()` to instead use `FlowAnnotations.GetXAnnotation()` (except for when we warn on annotations on unsupported types)? Or at least a doc comment on it pointing to the `FlowAnnotations` for most use cases.

Also, I think we might still have extra warnings about DAM annotation mismatch on base/override method generic parameters, and property definition and accessor.

https://github.com/dotnet/runtime/blob/7cd266e3ad282d94e19664c6078ee699b9a5cc3e/src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs#L223-L246

https://github.com/dotnet/runtime/blob/7cd266e3ad282d94e19664c6078ee699b9a5cc3e/src/tools/illink/src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs#L248-L254